### PR TITLE
Supply API request data to context

### PR DESCRIPTION
**Pull request by zimnx**

### CODE REVIEW

**Comment by zimnx** [comment]
Tests added

File: server/api.go [hunk 1]
```diff
@@ -184,10 +184,11 @@ func fillInContext(context middleware.Context, db db.DB,
 	r *http.Request, w http.ResponseWriter,
 	s *schema.Schema, p martini.Params, sync sync.Sync,
 	identityService middleware.IdentityService,
-	queue *job.Queue) {
+	queue *job.Queue, requestData map[string]interface{}) {
 	context["path"] = r.URL.Path
 	context["http_request"] = r
 	context["http_response"] = w
+	context["request_data"] = requestData
```
Comment (p-kozlowski) [inline server/api.go]: [there are places](https://github.com/cloudwan/gohan/blob/master/server/resources/resource_management.go#L773) where the dataMap is modified, so the content of `context["request_data"]` may be modified after initial assignment. is it OK?
if not, we'd need a deep copy

File: server/server_test.go [hunk 2]
```diff
@@ -896,6 +896,24 @@ var _ = Describe("Server package test", func() {
 			testURL("POST", responderPluralURL, adminTokenID, responder, http.StatusCreated)
 		})
 
+		It("Request data not available in context on GET action", func() {
+			testURL("GET", responderPluralURL+"/r1", adminTokenID, nil, http.StatusOK)
+		})
+
+		It("Request data not available in context on DELETE action", func() {
```
Comment (p-kozlowski) [inline server/server_test.go]: please add tests for create and update, too

Comment (zimnx) [reply server/server_test.go]: other existing tests covers create and update

File: server/api.go (hunk 1 above)
Comment (zimnx) [reply server/api.go]: Deepcopy added

**Comment by morrisson** [comment]
`extension.md` should be updated.

**Comment by zimnx** [comment]
@morrisson I've added documentation.

**Pull request merged**